plugins/catalog/src/data/filters.ts

```diff
@@ -56,24 +54,11 @@ export const filterGroups: CatalogFilterGroup[] = [
   },
 ];
 
-type ResolverFunction = ({
-  catalogApi,
-  isStarredEntity,
-}: {
-  catalogApi: CatalogApi;
-  isStarredEntity: (entity: Entity) => boolean;
-}) => Promise<Entity[]>;
-
-export const dataResolvers: Record<FilterGroupItem, ResolverFunction> = {
-  [FilterGroupItem.OWNED]: async () => [],
-  [FilterGroupItem.ALL]: async ({ catalogApi }) => {
-    return catalogApi.getEntities();
-  },
-  [FilterGroupItem.STARRED]: async ({ catalogApi, isStarredEntity }) => {
-    const allEntities = await catalogApi.getEntities();
-
-    return allEntities.filter(entity => isStarredEntity(entity));
-  },
+export const entityFilters = {
```

Comment:
This one would benefit from being typed. Just looking at this object it isn't very clear what it does or how it works.

Reply:
Good improvement, should also declare on this one that `export const entityFilters: Record<string, (options: EntityFilterOptions) => EntityFilter>` because that really helps both the reader and the typechecker.

As a matter of fact, it may be clearer to do as follows to get rid of the nesting:

```ts
export type EntityContext = { isStarred: boolean };

export type EntityFilter = (entity: Entity, context: EntityContext) => boolean;

export const entityFilters: Record<string, EntityFilter> = {
  [EntityFilterType.OWNED]: () => false,
  [EntityFilterType.ALL]: () => true,
  [EntityFilterType.STARRED]: (entity, { isStarred }) => isStarred,
};
```

What do you think? @benjdlambert as well ^ But maybe do that as a later refactor.

Reply:
Sure - look's good to me!